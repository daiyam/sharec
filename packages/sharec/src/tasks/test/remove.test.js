const { fixtures } = require('testUtils')
const { vol } = require('memfs')
const remove = require('../remove')

describe('tasks > remove >', () => {
  const packageJsonBaseRemoveFxt = fixtures('package/04-base-remove', 'json')
  const babelListedValuesFxt = fixtures('babel/json/04-listed-values', 'json')
  const eslintParserOptionsOperationsFxt = fixtures(
    'eslint/json/02-parser-options-operations',
    'json',
  )
  const eslintParserOptionsOperationsFxtYaml = fixtures(
    'eslint/yaml/02-parser-options-operations',
  )

  beforeEach(() => {
    vol.reset()
  })

  it('should remove configuration from the target project and delete all empty', async () => {
    expect.assertions(3)

    const packageJson = {
      sharec: {
        injected: true,
      },
    }
    const dir = {
      '/target/package.json': JSON.stringify(packageJson),
      '/target/.eslintrc': JSON.stringify(
        eslintParserOptionsOperationsFxt.result,
      ),
      '/target/.babelrc': JSON.stringify(babelListedValuesFxt.result),
      '/configuration-package/configs/.eslintrc': JSON.stringify(
        eslintParserOptionsOperationsFxt.new,
      ),
      '/configuration-package/configs/.babelrc': JSON.stringify(
        babelListedValuesFxt.new,
      ),
    }
    vol.fromJSON(dir)

    await remove({
      targetPath: '/target',
      configsPath: '/configuration-package',
    })

    expect(
      JSON.parse(vol.readFileSync('/target/package.json')),
    ).toMatchSnapshot()
    expect(JSON.parse(vol.readFileSync('/target/.eslintrc', 'utf8'))).toEqual(
      eslintParserOptionsOperationsFxt.restored,
    )
    expect(JSON.parse(vol.readFileSync('/target/.babelrc'))).toEqual(
      babelListedValuesFxt.restored,
    )
  })

  it('should remove configuration from package.json in the target project', async () => {
    expect.assertions(1)

    const dir = {
      '/target/package.json': JSON.stringify(packageJsonBaseRemoveFxt.result),
      '/configuration-package/configs/package.json': JSON.stringify(
        packageJsonBaseRemoveFxt.new,
      ),
    }
    vol.fromJSON(dir)

    await remove({
      targetPath: '/target',
      configsPath: '/configuration-package',
    })

    expect(
      JSON.parse(vol.readFileSync('/target/package.json')),
    ).toMatchSnapshot()
  })

  it('should remove configuration from all files in the target project', async () => {
    expect.assertions(3)

    const dir = {
      '/target/package.json': JSON.stringify(packageJsonBaseRemoveFxt.result),
      '/target/.eslintrc.yml': eslintParserOptionsOperationsFxtYaml.result,
      '/target/.babelrc': JSON.stringify(babelListedValuesFxt.result),
      '/configuration-package/configs/package.json': JSON.stringify(
        packageJsonBaseRemoveFxt.new,
      ),
      '/configuration-package/configs/.eslintrc.yml':
        eslintParserOptionsOperationsFxtYaml.new,
      '/configuration-package/configs/.babelrc': JSON.stringify(
        babelListedValuesFxt.new,
      ),
    }
    vol.fromJSON(dir)

    await remove({
      targetPath: '/target',
      configsPath: '/configuration-package',
    })

    expect(vol.readFileSync('/target/.eslintrc.yml', 'utf8')).toEqual(
      eslintParserOptionsOperationsFxtYaml.restored,
    )
    expect(JSON.parse(vol.readFileSync('/target/.babelrc'))).toEqual(
      babelListedValuesFxt.restored,
    )
    expect(JSON.parse(vol.readFileSync('/target/package.json'))).toEqual(
      packageJsonBaseRemoveFxt.restored,
    )
  })

  it('should throw an error if configuration package was not found', async () => {
    expect.assertions(1)

    const dir = {
      '/target/package.json': JSON.stringify(
        {
          sharec: {
            injected: true,
          },
        },
        null,
        2,
      ),
    }
    vol.fromJSON(dir, '/')

    try {
      await remove({
        targetPath: '/target',
        configsPath: '/configuration-package',
      })
    } catch (err) {
      expect(err.message).toContain('ENOENT')
    }
  })
})
