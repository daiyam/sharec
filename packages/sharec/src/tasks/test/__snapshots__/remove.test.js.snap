// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tasks > remove > should remove configuration from all files in the target project 1`] = `
"env:
  node: false
extends: standard
globals:
  baz: true
rules:
  max-len:
    - warn
    - 80
    - 2
  sort-imports: 0
"
`;

exports[`tasks > remove > should remove configuration from all files in the target project 2`] = `
Object {
  "env": Object {
    "test": Object {
      "presets": Array [
        "stage-3",
      ],
    },
  },
  "plugins": Array [
    Array [
      "syntax-dynamic-import",
      Object {
        "foo": "baz",
      },
    ],
  ],
  "presets": Array [
    Array [
      "env",
      Object {
        "modules": false,
        "targets": Object {
          "browsers": Array [
            "> 1%",
            "last 2 versions",
            "not ie > 0",
          ],
        },
      },
    ],
  ],
}
`;

exports[`tasks > remove > should remove configuration from all files in the target project 3`] = `
Object {
  "config": Object {
    "commitizen": Object {
      "path": "cz-conventional-changelog",
    },
  },
  "devDependencies": Object {
    "@commitlint/cli": "^7.6.1",
    "cz-conventional-changelog": "^2.1.0",
    "eslint-config-prettier": "^4.2.0",
  },
  "eslintConfig": Object {
    "env": Object {
      "node": false,
    },
    "extends": "standard",
    "globals": Object {
      "baz": true,
    },
    "rules": Object {
      "max-len": Array [
        "warn",
        80,
        2,
      ],
      "sort-imports": 0,
    },
  },
  "eslintIgnore": Array [
    "/node_modules",
  ],
  "husky": Object {
    "hooks": Object {
      "commit-msg": "commitlint -E",
      "pre-commit": "lint-staged",
    },
  },
  "lint-staged": Object {
    "src/**/*.{js,json}": Array [
      "eslint --fix",
      "prettier --write",
      "git add",
    ],
  },
  "name": "awesome-project",
  "prettier": Object {
    "semi": false,
    "singleQuote": true,
    "trailingComma": "all",
  },
  "scripts": Object {
    "commit": "git-cz",
    "lint": "eslint 'src/**/*.{jsx,json}'",
    "lint-staged": "lint-staged",
  },
}
`;

exports[`tasks > remove > should remove configuration from package.json in the target project 1`] = `
Object {
  "config": Object {
    "commitizen": Object {
      "path": "cz-conventional-changelog",
    },
  },
  "devDependencies": Object {
    "@commitlint/cli": "^7.6.1",
    "cz-conventional-changelog": "^2.1.0",
    "eslint-config-prettier": "^4.2.0",
  },
  "eslintConfig": Object {
    "env": Object {
      "node": false,
    },
    "extends": "standard",
    "globals": Object {
      "baz": true,
    },
    "rules": Object {
      "max-len": Array [
        "warn",
        80,
        2,
      ],
      "sort-imports": 0,
    },
  },
  "eslintIgnore": Array [
    "/node_modules",
  ],
  "husky": Object {
    "hooks": Object {
      "commit-msg": "commitlint -E",
      "pre-commit": "lint-staged",
    },
  },
  "lint-staged": Object {
    "src/**/*.{js,json}": Array [
      "eslint --fix",
      "prettier --write",
      "git add",
    ],
  },
  "name": "awesome-project",
  "prettier": Object {
    "semi": false,
    "singleQuote": true,
    "trailingComma": "all",
  },
  "scripts": Object {
    "commit": "git-cz",
    "lint": "eslint 'src/**/*.{jsx,json}'",
    "lint-staged": "lint-staged",
  },
}
`;

exports[`tasks > remove > should remove configuration from the target project and delete all empty 1`] = `Object {}`;

exports[`tasks > remove > should remove configuration from the target project and delete all empty 2`] = `
Object {
  "env": Object {
    "node": false,
  },
  "extends": "standard",
  "globals": Object {
    "baz": true,
  },
  "rules": Object {
    "max-len": Array [
      "warn",
      80,
      2,
    ],
    "sort-imports": 0,
  },
}
`;

exports[`tasks > remove > should remove configuration from the target project and delete all empty 3`] = `
Object {
  "env": Object {
    "test": Object {
      "presets": Array [
        "stage-3",
      ],
    },
  },
  "plugins": Array [
    Array [
      "syntax-dynamic-import",
      Object {
        "foo": "baz",
      },
    ],
  ],
  "presets": Array [
    Array [
      "env",
      Object {
        "modules": false,
        "targets": Object {
          "browsers": Array [
            "> 1%",
            "last 2 versions",
            "not ie > 0",
          ],
        },
      },
    ],
  ],
}
`;
